File: schemaorg/returnaction.go
package schemaorg

import "encoding/json"

// ReturnAction see : https://schema.org/ReturnAction
type ReturnAction struct {

typeContext

TransferAction

// Recipient see : /recipient
// A sub property of participant. The participant who is at the receiving end of the action.
Recipient interface{} `json:"recipient"` // types : Audience ContactPoint Organization Person

}

func (v *ReturnAction) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "ReturnAction"

	return json.Marshal(v)
}

File: schemaorgphp/classes/return_action.php
<?php

class ReturnAction extends TransferAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'ReturnAction';
	
	/**
	 * A sub property of participant. The participant who is at the receiving end of the action.
	 * see : https://schema.org/recipient
	 * @var \Audience|\Audience[]|\ContactPoint|\ContactPoint[]|\Organization|\Organization[]|\Person|\Person[]
	 */
	public var $recipient;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'ReturnAction'
		);
		
		$serialized = so_json_serialize( $this->recipient );
		if ( ! empty( $serialized ) ) {
			$out['recipient'] = $serialized;
		}
		
		return $out;
	}
}
